bridge/node/js/graph.ts

```diff
@@ -1,25 +1,110 @@
+
+    getGraphCapacity(): Promise<number> {
+        return graphsdkModule.getGraphCapacity(this.handle);
+    }
+
+    getGraphSize(): Promise<number> {
+        return graphsdkModule.getGraphSize(this.handle);
+    }
+
+    containsUserGraph(dsnpUserId: number): Promise<boolean> {
+        return graphsdkModule.containsUserGraph(this.handle, dsnpUserId);
+    }
+
+    getGraphUsersCount(): Promise<number> {
+        return graphsdkModule.getGraphUsersCount(this.handle);
+    }
+
+    removeUserGraph(dsnpUserId: number): Promise<void> {
+        return graphsdkModule.removeUserGraph(this.handle, dsnpUserId);
+    }
+
+    importUserData(payload: [ImportBundle]): Promise<void> {
+        return graphsdkModule.importUserData(this.handle, payload);
+    }
+
+    exportUpdates(): Promise<Update> {
+        return graphsdkModule.exportUpdates(this.handle);
+    }
+
+    getConnectionsForUserGraphUpdates(dsnpUserId: number, schemaId: string, includePending: boolean): Promise<[DsnpGraphEdge]> {
+        return graphsdkModule.getConnectionsForUserGraphUpdates(this.handle, dsnpUserId, schemaId, includePending);
     }
 
-    public getGraphConfig(environment: EnvironmentInterface): Config {
+    applyActions(actions: [Action]): Promise<void> {
+        return graphsdkModule.applyActions(this.handle, actions);
+    }
+
+    forceCalculateGraphs(dsnpUserId: number): Promise<Update> {
+        return graphsdkModule.forceCalculateGraphs(this.handle, dsnpUserId);
+    }
+
+    getConnectionsWithoutKeys(): Promise<[number]> {
+        return graphsdkModule.getConnectionsWithoutKeys(this.handle);
+    }
+
+    getOneSidedPrivateFriendshipConnections(dsnpUserId: number): Promise<[DsnpGraphEdge]> {
+        return graphsdkModule.getOneSidedPrivateFriendshipConnections(this.handle, dsnpUserId);
+    }
+
+    getPublicKeys(dsnpUserId: number): Promise<[DsnpPublicKey]> {
+        return graphsdkModule.getPublicKeys(this.handle, dsnpUserId);
+    }
+
+    deserializeDsnpKeys(keys: DsnpKeys): Promise<[DsnpPublicKey]> {
+        return graphsdkModule.deserializeDsnpKeys(keys);
+    }
+
+    public getGraphConfig(environment: EnvironmentInterface): Promise<Config> {
         return graphsdkModule.getGraphConfig(environment);
     }
-    
+
+    // Finalizer to free the graph state
+    private registerFinalizer(): void {
+        const finalizer = () => {
+            this.freeGraphState();
+        };
+
+        // Register the finalizer
+        if (typeof FinalizationRegistry !== "undefined") {
+            const registry = new FinalizationRegistry<Graph>(() => {
+                finalizer();
+            });
+            registry.register({}, this);
+        } else if (typeof process !== "undefined" && typeof process.on === "function") {
+            process.on("exit", finalizer);
+        } else {
+            console.warn("Unable to register finalizer. Memory may not be freed correctly.");
+        }
+    }
+
+    public async freeGraphState(): Promise<void> {
```

Comment:
I wonder if, since we're using a registered finalizer, we should make this function private--otherwise you could call `freeGraphState` and then this object would be left with a dangling handle.

Reply:
Ideally it wont happen given state will be remove from hashmap and calling function will just get an error, but will hide it

Reply:
Apparently finalizer wont free the state tried few examples, for now we will explicitly free the state and mention that in documentation

Reply:
I'm not sure I understand... so are you saying the registered finalizer doesn't work? If so, should we just remove that?

Reply:
Yes it does not work, i think it would need to know the actual State object to be able to track it and call garbage collection on it, but we hide the state and manage it on rust side, so we have to explicitly garbage collect hidden state

Reply:
I removed the finalizer for now, we would call free state explicitly as a start?

Reply:
That seems fine for now. I'll put a story in the backlog to investigate getting the finalizer to work.